### PR TITLE
Add custom font and scripts notice

### PR DESCRIPTION
**Pull request by qwerty287**
Add instructions on how to use a custom font on install page and on how to add a custom script (closes https://github.com/LycheeOrg/Lychee/issues/1255 by adding an FAQ).

### CODE REVIEW

File: docs/faq.md [hunk 1]
```diff
@@ -92,6 +92,64 @@ Add the following custom CSS to your `user.css` or via the settings menu:
 }
 ```
 
+### How to use a custom font on the landing page?
+
+For example, if you want to change it to Roboto, add the following custom CSS to the your `user.css` or via the settings menu:
```
Comment (nagmat84) [inline docs/faq.md:97]: To me the flow of thought and the introduction feels slightly strange. The initial sentence starts with a specific example, but then the CSS has a placeholder which is generic. Either make it specific or generic, but be consistent about it. Personally, I would start with a generic example.

```suggestion
In order to use a specific font on the landing page which does not rely on some font provided by the client-side browser, you have two options:

 1. self-host the necessary set of font files as part of your Lychee installation
 2. use an external font-hosting service such as Google Fonts

Both options require to tweak the custom `user.css` via the settings menu.

#### Option 1: Self-hosted font files

Let's assume that your favorite font is called `'My Font'`. You need a set of the font files in WOFF2 format with one font file per used font weight and style. The landing page uses 5 different font weights (200 = thin, 300 = light, 400 = normal, 500 = thick, 700 = bold) all in normal (i.e. upright) style.

The necessary CSS consists of two parts: firstly, the font face and its different weights are declared via the `@font-face`-directive; secondly, the declared font needs to be used by the styles for the page. A template CSS code for `user.css` looks like this.
```

File: docs/faq.md [hunk 2]
```diff
@@ -92,6 +92,64 @@ Add the following custom CSS to your `user.css` or via the settings menu:
...
+
+```css
+@font-face {
+    font-family: "Roboto";
```
Comment (nagmat84) [inline docs/faq.md:101]:
```suggestion
    font-family: "My Font";
```

File: docs/faq.md [hunk 3]
```diff
@@ -92,6 +92,64 @@ Add the following custom CSS to your `user.css` or via the settings menu:
...
+    font-style: normal;
+    font-weight: 200;
+    src: local(""), url("<font URL for weigth 200>") format("woff2");
```
Comment (nagmat84) [inline docs/faq.md:104]:
```suggestion
    src: local(""), url("<font URL for weight 200>") format("woff2");
```

File: docs/faq.md [hunk 4]
```diff
@@ -92,6 +92,64 @@ Add the following custom CSS to your `user.css` or via the settings menu:
...
+    font-style: normal;
+    font-weight: 700;
+    src: local(""), url("<font URL for weigth 700>") format("woff2");
+}
+
+#logo h1 span,
+#logo h1,
+#intro,
+.menu .menu-item {
+    font-family: "Roboto", sans-serif !important;
```
Comment (nagmat84) [inline docs/faq.md:135]:
```suggestion
    font-family: "My Font", sans-serif !important;
```

File: docs/faq.md [hunk 5]
```diff
@@ -92,6 +92,64 @@ Add the following custom CSS to your `user.css` or via the settings menu:
...
+}
+
+#intro .subtitle,
+#logo .subtitle {
+    margin-top: 0 !important;
+}
```
Comment (nagmat84) [inline docs/faq.md:141]: Is this required or related to using an alternative font?

File: docs/faq.md [hunk 6]
```diff
@@ -92,6 +92,64 @@ Add the following custom CSS to your `user.css` or via the settings menu:
...
+```
+
+Probably, you have to customize this:
+* Replace the placeholders with your actual URLs.
+* To use a different font, just change the URLs to match your font and the word `Roboto` to your font's name.
+* If you want to use a hosted service like Google Fonts, replace the `@font-face` blocks with `@import url("https://fonts.googleapis.com/css?family=Roboto:200,300,400,500,700");` and change `Roboto` to your font's name on Google Fonts.
```
Comment (nagmat84) [inline docs/faq.md:147]: I don't understand the term "probably". It is for sure that the placeholders in the template must be replaced.

```suggestion

#### Option 2: Using an external font-hosting service

If you want to use an external font-hosting service, replace the `@font-face` blocks of the template above with an `@import`-directive which points to the externally hosted font. The name of the font-family must be replaced by the font name as documented by the external font-service.

For example, let's assume you want to restore the old look-and-feel of the landing page which used Google's Roboto font. Then the CSS has to look like

```css
@import url("https://fonts.googleapis.com/css?family=Roboto:200,300,400,500,700");

#logo h1 span,
#logo h1,
#intro,
.menu .menu-item {
    font-family: "Roboto", sans-serif !important;
}
```

```

Sorry, my suggestion misses the final three backticks, because Github becomes confused with them.

**Review by nagmat84 — COMMENTED** [review]
Just my 2ct.

File: docs/faq.md (hunk 5 above)
Comment (nagmat84) [reply docs/faq.md:152]: I assumed so. But then I recommend to keep it out of the generic template and present it as part of the template for Google Roboto. Also add a remark about it which explains it.

File: docs/faq.md [hunk 7]
```diff
@@ -92,6 +92,88 @@ Add the following custom CSS to your `user.css` or via the settings menu:
...
+    font-family: "My Font", sans-serif !important;
+}
+
+#intro .subtitle,
+#logo .subtitle {
+    margin-top: 0 !important;
+}
+```
+
+#### Option 2: Using an external font-hosting service
```
Comment (nagmat84) [inline docs/faq.md:155]: I am not 100% sure, but I believe we need a blank line after the caption otherwise the rendering fails. (Of course it ways me how introduced that error due to my suggestion.)

```suggestion
#### Option 2: Using an external font-hosting service

```

File: docs/faq.md [hunk 8]
```diff
@@ -92,6 +92,88 @@ Add the following custom CSS to your `user.css` or via the settings menu:
...
+If you want to use an external font-hosting service, replace the `@font-face` blocks of the template above with an `@import`-directive which points to the externally hosted font. The name of the font-family must be replaced by the font name as documented by the external font-service.
+For example, let's assume you want to restore the old look-and-feel of the landing page which used Google's Roboto font. Then the CSS has to look like
```
Comment (nagmat84) [inline docs/faq.md:157]: Same think here.

```suggestion
For example, let's assume you want to restore the old look-and-feel of the landing page which used Google's Roboto font. Then the CSS has to look like

```

**Review by nagmat84 — COMMENTED** [review]
Added some blank lines, because I assume we need them.

**Review by nagmat84 — APPROVED** [review]
LGTM. :+1:

File: docs/faq.md (hunk 5 above)
Comment (nagmat84) [reply docs/faq.md:172]: I'm fine with it.

File: docs/faq.md [hunk 9]
```diff
@@ -92,6 +92,99 @@ Add the following custom CSS to your `user.css` or via the settings menu:
...
+#intro .subtitle,
+#logo .subtitle {
+    margin-top: 0 !important;
+}
+```
+
+### How to add custom scripts?
+
+If you want to add custom scripts, for example to add an analytics service, you can add the script code to the `additional_footer_text` setting. For example, adding `<script>alert('Hello World!');</script>` to it will display an alert dialog with `Hello World!` in it every time you load the page.
+
```
Comment (kamil4) [inline docs/faq.md:187]: Why is this part of this PR?

File: docs/faq.md [hunk 10]
```diff
@@ -92,6 +92,99 @@ Add the following custom CSS to your `user.css` or via the settings menu:
...
+```
+
+Some fonts may require additional changes to improve the margin of the subtitles. One of these fonts is Roboto. Therefore, you can add following:
+
+```css
+#intro .subtitle,
+#logo .subtitle {
+    margin-top: 0 !important;
+}
+```
```
Comment (kamil4) [inline docs/faq.md:182]: This CSS section occurs twice before the explanation provided here; as such the explanation seems too late. Can we move it to the discussion of option 1?

**Review by kamil4 — CHANGES_REQUESTED** [review]

File: docs/faq.md [hunk 11]
```diff
@@ -92,6 +92,80 @@ Add the following custom CSS to your `user.css` or via the settings menu:
...
+
+```css
+@import url("https://fonts.googleapis.com/css?family=Roboto:200,300,400,500,700");
+#logo h1 span,
+#logo h1,
+#intro,
+.menu .menu-item {
+    font-family: "Roboto", sans-serif !important;
+}
+```
```
Comment (kamil4) [inline docs/faq.md:163]: The example is useful but the details need to be updated since we are back to using Roboto in the landing page. Maybe simply give an example of using another Google font (like the one that we used in error pages, whatever it's name was?).

**Review by kamil4 — CHANGES_REQUESTED** [review]

**Review by kamil4 — APPROVED** [review]
